File: examples/cli_flags.js
const args = process.argv.slice(2)

let foo = 'default value'
let qux = false

for (let i = 0; i < args.length; i++) {
  const arg = args[i]
  if (arg === '--foo' || arg === '-foo') {
    foo = args[i + 1]
    i++
  } else if (arg === '--qux' || arg === '-qux') {
    qux = true
  }
}

console.log('foo:', foo)
console.log('qux:', qux)

File: examples/cli_flags.cpp
#include <algorithm>
#include <chrono>
#include <cstdlib>
#include <cstring>
#include <exception>
#include <fstream>
#include <functional>
#include <future>
#include <iostream>
#include <map>
#include <numeric>
#include <regex>
#include <sstream>
#include <stdexcept>
#include <string>
#include <thread>
#include <vector>

int main(int argc, char **argv) { const char *foo = "default value"; bool qux = false; for (int i = 1; i < argc; i++) { if (strcmp(argv[i], "--foo") == 0 && i + 1 < argc) foo = argv[++i]; else if (strcmp(argv[i], "--qux") == 0) qux = true; } printf("foo: %s\n", foo); printf("qux: %s\n", qux ? "true" : "false"); return 0; }
